cmd/main.go

```diff
@@ -0,0 +1,128 @@
+)
+
+func Run() {
+
+	promlogConfig := &promlog.Config{}
+
+	flag.AddFlags(kingpin.CommandLine, promlogConfig)
+	kingpin.Version(version.Print("json_exporter"))
+	kingpin.HelpFlag.Short('h')
+	kingpin.Parse()
+	logger := promlog.New(promlogConfig)
+
+	level.Info(logger).Log("msg", "Starting json_exporter", "version", version.Info()) //nolint:errcheck
+	level.Info(logger).Log("msg", "Build context", "build", version.BuildContext())    //nolint:errcheck
+
+	level.Info(logger).Log("msg", "Loading config file", "file", *configFile) //nolint:errcheck
+	config, err := config.LoadConfig(*configFile)
+	if err != nil {
+		level.Error(logger).Log("msg", "Error loading config", "err", err) //nolint:errcheck
+		os.Exit(1)
+	}
+	configJson, err := json.Marshal(config)
+	if err != nil {
+		level.Error(logger).Log("msg", "Failed to marshal config to JOSN", "err", err) //nolint:errcheck
+	}
+	level.Info(logger).Log("msg", "Loaded config file", "config", string(configJson)) //nolint:errcheck
+
+	if *configCheck {
+		os.Exit(0)
+	}
+
+	http.Handle("/metrics", promhttp.Handler())
+	http.HandleFunc("/probe", func(w http.ResponseWriter, req *http.Request) {
+		probeHandler(w, req, logger, config)
+	})
+	if err := http.ListenAndServe(*listenAddress, nil); err != nil {
+		level.Error(logger).Log("msg", "failed to start the server", "err", err) //nolint:errcheck
+	}
+}
+
+func probeHandler(w http.ResponseWriter, r *http.Request, logger log.Logger, config config.Config) {
+
+	ctx, cancel := context.WithTimeout(r.Context(), time.Duration(config.Global.TimeoutSeconds*float64(time.Second)))
+	defer cancel()
+	r = r.WithContext(ctx)
+
+	registry := prometheus.NewPedanticRegistry()
+
+	metrics, err := internal.CreateMetricsList(registry, config)
+	if err != nil {
+		level.Error(logger).Log("msg", "Failed to create metrics list from config", "err", err) //nolint:errcheck
+	}
+
+	probeSuccessGauge := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "probe_success",
+		Help: "Displays whether or not the probe was a success",
+	})
+	probeDurationGauge := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "probe_duration_seconds",
+		Help: "Returns how long the probe took to complete in seconds",
+	})
+
+	target := r.URL.Query().Get("target")
+	if target == "" {
+		http.Error(w, "Target parameter is missing", http.StatusBadRequest)
+		return
+	}
+
+	start := time.Now()
+	registry.MustRegister(probeSuccessGauge)
+	registry.MustRegister(probeDurationGauge)
+
+	data, err := internal.FetchJson(ctx, logger, target, config.Headers)
+	if err != nil {
+		level.Error(logger).Log("msg", "Failed to fetch JSON response", "err", err) //nolint:errcheck
+		duration := time.Since(start).Seconds()
+		level.Error(logger).Log("msg", "Probe failed", "duration_seconds", duration) //nolint:errcheck
+	} else {
+		internal.Scrape(logger, metrics, data)
+
+		duration := time.Since(start).Seconds()
+		probeDurationGauge.Set(duration)
```

Comment:
Because the scrape and probe is 1:1, we don't really need this as it's covered by the `scrape_duraiton_seconds` in Prometheus.

Reply:
Removed!